cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py

```diff
@@ -34,24 +35,31 @@ def request_server_certificates(tls):
       'certificates.server.cert.available')
 def install_load_balancer(apiserver, tls):
     ''' Create the default vhost template for load balancing '''
-    hookenv.open_port(hookenv.config('port'))
-    services = apiserver.services()
-
+    # Get the tls paths from the layer data.
     layer_options = layer.options('tls-client')
-    certificates_directory = layer_options.get('certificates-directory')
-    server_certificate = os.path.join(certificates_directory,
-                                      'server.crt')
-    server_key = os.path.join(certificates_directory, 'server.key')
-    nginx.configure_site(
-            'apilb',
-            'apilb.conf',
-            server_name='_',
-            services=services,
-            port=hookenv.config('port'),
-            server_certificate=server_certificate,
-            server_key=server_key,
-    )
-    hookenv.status_set('active', 'Loadbalancer ready.')
+    server_cert_path = layer_options.get('server_certificate_path')
+    cert_exists = server_cert_path and os.path.isfile(server_cert_path)
+    server_key_path = layer_options.get('server_key_path')
+    key_exists = server_key_path and os.path.isfile(server_key_path)
+
+    if cert_exists and key_exists:
```

Comment:
Nit: It would be good to insert a comment here why we are messing with the ownership of the certificates